proxy/proxy.go

```diff
@@ -186,6 +224,12 @@ func (p *Proxy) UpdateServices(services []*Service) error {
 
 // Close cleans up the Proxy by closing any remaining open connections.
 func (p *Proxy) Close() error {
+	for _, s := range p.services {
```

Comment:
We probably don't want to _not_ close the other pricers if one of them fails.
What we usually do in such a case is to declare a `var returnErr error` and set that if one of the errors is non-nil and then return that variable in the end. That way we only return the last error but at least close all of the items in the loop. Can add a log statement to make sure consecutive errors are at least seen.

Reply:
ah yes! good catch